File: csharp/raindrops/Raindrops.cs
using System;

public static class Raindrops {
    public static string Convert (int number) {
        bool fac3 = number % 3 == 0;
        bool fac5 = number % 5 == 0;
        bool fac7 = number % 7 == 0;
        if (fac3 || fac5 || fac7) {
            string rain = "";
            if (fac3) {
                rain += "Pling";
            }
            if (fac5) {
                rain += "Plang";
            }
            if (fac7) {
                rain += "Plong";
            }
            return rain;
        }
        return number.ToString ();
    }
}

File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
)

const testVersion = 3

// Convert converts a number into a string based on the factors present in it
func Convert(num int) string {
	var ppp string
	factors := []int{3, 5, 7}
	raindrop := []string{"Pling", "Plang", "Plong"}
	for i := 0; i < len(factors); i++ {
		if num%factors[i] == 0 {
			ppp += raindrop[i]
		}
	}
	if len(ppp) == 0 {
		return fmt.Sprintf("%d", num)
	}
	return ppp
}
